src/include/stir/numerics/MatrixFunction.inl

```diff
@@ -106,6 +106,44 @@ namespace detail
 	  retval[i] += m[i][j]*vec[j];
       }
   }
+
+  template <class elemT, class vecT>
+  inline
+  void
+  matrix_matrixT_multiply_help(vecT& retval, const Array<2,elemT>& m, const vecT& vec)
+  {
+      assert(m.is_regular());
+      if (m.size()==0)
+      { return; }
+
+      const int m_min_row = m.get_min_index();
+      const int m_max_row = m.get_max_index();
+      const int m_min_col = m[m_min_row].get_min_index();
+      const int m_max_col = m[m_min_row].get_max_index();
+      // make sure matrices are conformable for multiplication
+      assert(vec.get_min_index() == m_min_col);
+      assert(vec.get_max_index() == m_max_col);
+
+      float cov = 0.f;
+      float norm_factor = 1.f / m_max_row;
+      int k = m_min_row;
+
+      for(int i=m_min_col; i<=m_max_col; ++i)
+      {
+          for(int j= m_min_col; j<= i; ++j)
+          {
+              k = m_min_row;
+              cov = m[k][i] * m[k][j];
+              for (++k; k<= m_max_row; ++k)
```

Comment:
@KrisThielemans 
I would definitely consider caching this loop. 
We could cache half the CoV values and never loop over it again.